### PR TITLE
Introducing type hints

### PR DESCRIPTION
**Pull request by shacharoo**
Added  the minimal code needed for typehint validation.
I also removed some whitespaces in tox.ini - it seems that tox doesn't remove them (for instance, py34 install env was called `py34- install`).

### CODE REVIEW

File: src/pydocstyle/cli.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def run_pydocstyle():
         return ReturnCode.invalid_options
 
     count = 0
-    for error in errors:
+    for error in errors:  # type: ignore
```
Comment (Nurdok) [inline src/pydocstyle/cli.py]: Why?

File: src/pydocstyle/parser.py [hunk 2]
```diff
@@ -102,7 +102,7 @@ def __str__(self):
 class Module(Definition):
     """A Python source code module."""
 
-    _fields = ('name', '_source', 'start', 'end', 'decorators', 'docstring',
+    _fields = ('name', '_source', 'start', 'end', 'decorators', 'docstring',  # type: ignore
```
Comment (Nurdok) [inline src/pydocstyle/parser.py]: Why?

File: src/tests/test_cases/sections.py [hunk 3]
```diff
@@ -48,7 +48,7 @@ def no_underline():
 
 @expect(_D213)
 @expect("D407: Missing dashed underline after section ('Returns')")
-def no_underline():
+def no_underline_and_no_description():
```
Comment (Nurdok) [inline src/tests/test_cases/sections.py]: Why?

File: src/pydocstyle/parser.py (hunk 2 above)
Comment (shacharoo) [reply src/pydocstyle/parser.py]: Overriding the `_fields` member of the base class:
```
py37-tests runtests: commands[1] | mypy --config-file=tox.ini src/
src\pydocstyle\parser.py:105: error: Incompatible types in assignment (expression has type "Tuple[str, str, str, str, str, str, str, str, str, str, str, str]", base class "Definition" defined the type as "Tuple[str, str, str, str, str, str, str, str, str]")
```

File: src/pydocstyle/cli.py (hunk 1 above)
Comment (shacharoo) [reply src/pydocstyle/cli.py]: Hiding the `error` variable defined above in `except IllegalConfiguration as error:`. I could rename it if you want.

File: src/tests/test_cases/sections.py (hunk 3 above)
Comment (shacharoo) [reply src/tests/test_cases/sections.py]: A function with the same name was defined above, my (historical) bad 😋

File: src/pydocstyle/cli.py (hunk 1 above)
Comment (shacharoo) [reply src/pydocstyle/cli.py]:
```
py37-tests runtests: commands[1] | mypy --config-file=tox.ini src/
src\pydocstyle\cli.py:52: error: Assignment to variable 'error' outside except: block
```

**Pull request merged**